src/GCDWheel.cs

```diff
@@ -269,51 +341,621 @@ public unsafe class GCDWheel {
+            
+            // in Castbar mode:
+            // draw the slidecast bar
+            if (conf.SlideCastEnabled){
+                DrawSlideCast(ui, sc_sv, sc_ev, go);
+                }
+
+            // in GCDBar mode:
+            // draw oGCDs and clips
+            if (!isCastBar && !(shortCastFinished && conf.HideAnimationLock)) {
+                float gcdTime = gcdTime_slidecastStart;
+                float gcdTotal = gcdTotal_slidecastEnd;
+
+                foreach (var (ogcd, (anlock, iscast)) in ogcds) {
+                    var isClipping = CheckClip(iscast, ogcd, anlock, gcdTotal, gcdTime);
+                    float ogcdStart = (conf.BarRollGCDs && gcdTotal - ogcd < 0.2f) ? 0 + barGCDClipTime : ogcd;
+                    float ogcdEnd = ogcdStart + anlock;
+
+                    // Ends next GCD
+                    if (conf.BarRollGCDs && ogcdEnd > gcdTotal) {
+                        ogcdEnd = gcdTotal;
+                        barGCDClipTime += ogcdStart + anlock - gcdTotal;
+                        //prevent red bar when we "clip" a hard-cast ability
+                        if (!isHardCast) {
+                            // create end vertex
+                            Vector2 clipEndVector = new(
+                                (int)(bar.CenterX + ((barGCDClipTime / gcdTotal) * bar.Width) - bar.HalfWidth),
+                                (int)(bar.CenterY + bar.HalfHeight)
+                            );
+                            // Draw the clipped part at the beginning
+                            ui.DrawRectFilledNoAA(bar.StartVertex, clipEndVector, conf.clipCol);
+                        }
+                    }
+
+                    Vector2 oGCDStartVector = new(
+                        (int)(bar.CenterX + ((ogcdStart / gcdTotal) * bar.Width) - bar.RawHalfWidth),
+                        (int)(bar.CenterY - bar.RawHalfHeight)
+                    );
+                    Vector2 oGCDEndVector = new(
+                        (int)(bar.CenterX + ((ogcdEnd / gcdTotal) * bar.Width) - bar.HalfWidth),
+                        (int)(bar.CenterY + bar.RawHalfHeight)
+                    );
+
+                    ui.DrawRectFilledNoAA(oGCDStartVector, oGCDEndVector, isClipping ? conf.clipCol : conf.anLockCol);
+                    if (!iscast && (!isClipping || ogcdStart > 0.01f)) {
+                        Vector2 clipPos = new(
+                            bar.CenterX + (ogcdStart / gcdTotal * bar.Width) - bar.RawHalfWidth,
+                            bar.CenterY - bar.RawHalfHeight + 1f
+                        );
+                        ui.DrawRectFilledNoAA(clipPos,
+                            clipPos + new Vector2(2f * ui.Scale, bar.Height - 2f),
+                            conf.ogcdCol);
+                    }
+                }
+            }
+
+            //in both modes:
+            //draw the queuelock (if enabled)
```

Comment:
Actually, that's all handled in the BarDecisionHelper (go.xxxx will only ever be true when it should be drawn).  Judgement call if you want to save the additional checks that happen inside DrawSlideCast and DrawQueueLock by checking if they're needed before we get there.

Judgement call if you want a few more lines of code to save a few conditional checks or not.  It wasn't possible at all until I redid BarDecisionHelper because we sometimes needed bits of the Queuelock even when it was off (after the Slidecast passed where the Queuelock should have been).

But now it is totally possible to check first, if you'd like to do so.

Reply:
More than saving the checks I was concerned that it was getting a queueLock by having, QueueLockEnabled = False and BarQueueLockWhenIdle = True.

Reply:
Yeah, that was a bug I noticed while working on the mount issue.  I left the check out in BarDecisonHelper ResetBar().  It’s fixed now:

```
Queue_Lock_Start = conf.QueueLockEnabled && conf.BarQueueLockWhenIdle ? 0.8f : 0f;   
Queue_VerticalBar = conf.QueueLockEnabled && conf.BarQueueLockWhenIdle;  
Queue_TopTriangle = conf.QueueLockEnabled && conf.BarQueueLockWhenIdle && conf.ShowQueuelockTriangles;  
```


I put the other check in as well.  Better safe than sorry.